qiita_pet/templates/artifact_ajax/processing_artifact.html

```diff
@@ -203,67 +213,140 @@
+  function load_parameter_gui(p_name, param_info, sel_artifacts_info, target, dflt_val, allow_change_optionals) {
+    // Create the parameter interface
+    var $rowDiv = $('<div>').addClass('row').addClass('form-group').appendTo(target);
+    // Replace the '_' by ' ' in the parameter name for readability
+    $('<label>').addClass('col-sm-2').addClass('col-form-label').text(p_name.replace('_', ' ') + ': ').appendTo($rowDiv).attr('for', p_name);
+    var $colDiv = $('<div>').addClass('col-sm-3').appendTo($rowDiv);
+
     var p_type = param_info[0];
     var allowed_types = param_info[1];
-    if (p_type == 'artifact') {
-      var sel = $('<select>').appendTo("#cmd-opts-div").attr('id', p_name).attr('name', p_name).addClass('required-parameter');
-      for(var key in sel_artifacts_info) {
-         if(allowed_types.indexOf(sel_artifacts_info[key].type) !== -1) {
-           sel.append($('<option>').attr('value', key).text(sel_artifacts_info[key].name));
-         }
+    var $inp;
+
+    if (p_type == 'artifact' || p_type.startsWith('choice') || p_type.startsWith('mchoice')) {
+      // The parameter type is an artifact or choice, the input type is a dropdown
+      $inp = $('<select>');
+      // show a dropdown menu with the
+      var options = [];
+      if (p_type.startsWith('choice') || p_type.startsWith('mchoice')) {
+        $.each(JSON.parse(p_type.split(':')[1]), function (idx, val) { options.push([val, val]); });
+
+        if (p_type.startsWith('mchoice')) {
+          $inp.attr('multiple', true);
+        }
       }
+      else {
+        // available artifacts of the given type
+        for(var key in sel_artifacts_info) {
+           if(allowed_types.indexOf(sel_artifacts_info[key].type) !== -1) {
+             options.push([key, sel_artifacts_info[key].name]);
+           }
+        }
+      }
+
+      $.each(options, function(idx, val) {
+        $inp.append($("<option>").attr('value', val[0]).text(val[1]));
+      });
     }
-    else if (p_type == 'integer') {
-      var inp = $('<input>').appendTo("#cmd-opts-div").attr('id', p_name).attr('name', p_name).attr('type', 'number').addClass('required-parameter');
+    else {
+      // The rest of parameter types are represented with an input
+      $inp = $('<input>');
+      // It just changes the type of input
+      if (p_type == 'integer') {
+        // For the integer type, show an input of type number
+        $inp.attr('type', 'number');
+      }
+      else if (p_type == 'float') {
+        // For the float type, show an input of type number, with a step of 0.001
+        $inp.attr('type', 'number').attr('step', 0.001);
+      }
+      else if (p_type == 'string') {
+        // For the float type, show an input of type text
+        $inp.attr('type', 'text');
+      }
+      else if (p_type == 'boolean') {
+        // For the boolean type, show an input of type checkbox
+        $inp.attr('type', 'checkbox');
+      }
+    }
+
+    if (dflt_val !== undefined) {
```

Comment:
All optional parameters have a default value - that's why they're optional (see the multiple available commands).